platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java

```diff
@@ -33,19 +33,19 @@ private AddressBookMetrics() {}
      * Register the metrics for the address book.
      *
      * @param metrics     the metrics engine
-     * @param addressBook the address book to register metrics for
+     * @param roster      the roster reflecting the address book to register metrics for
      * @param selfId      the ID of the node
      */
     public static void registerAddressBookMetrics(
-            @NonNull final Metrics metrics, @NonNull final AddressBook addressBook, @NonNull final NodeId selfId) {
+            @NonNull final Metrics metrics, @NonNull final Roster roster, @NonNull final NodeId selfId) {
 
         metrics.getOrCreate(new FunctionGauge.Config<>(Metrics.INFO_CATEGORY, "memberID", Long.class, selfId::id)
                 .withUnit("nodeID")
                 .withDescription("The node ID number of this member"));
 
-        metrics.getOrCreate(
-                new FunctionGauge.Config<>(Metrics.INFO_CATEGORY, "members", Integer.class, addressBook::getSize)
-                        .withUnit("count")
-                        .withDescription("total number of nodes currently in the address book"));
+        metrics.getOrCreate(new FunctionGauge.Config<>(
+                        Metrics.INFO_CATEGORY, "members", Integer.class, roster.rosterEntries()::size)
+                .withUnit("count")
+                .withDescription("total number of nodes currently in the address book"));
```

Comment:
Minor: s/address book/roster/

Also minor, might want to rename this class to `RosterMetrics` as well. That's what I originally did at https://github.com/hashgraph/hedera-services/pull/16026